projects/sunbird-video-player/src/lib/components/video-player/video-player.component.ts

```diff
@@ -230,8 +230,11 @@ export class VideoPlayerComponent implements AfterViewInit, OnInit, OnDestroy, O
     });
     events.forEach(event => {
       this.player.on(event, (data) => {
+        if (this.totalDuration === 0 && data.type === 'durationchange') {
```

Comment:
This is the generic method to dispatch all the events. This logic should not be present here. 

As discussed, of you need only duration verify the root where we are sending the duration & fix that code if you have any issue.